digest/src/core_api/rt_variable.rs

```diff
@@ -145,6 +156,54 @@ where
     }
 }
 
+impl<T> SerializableState for RtVariableCoreWrapper<T>
+where
+    T: VariableOutputCore + UpdateCore + SerializableState,
+    T::BlockSize: IsLess<U256>,
+    Le<T::BlockSize, U256>: NonZero,
+    T::SerializedStateSize: Add<U1>,
+    <T::SerializedStateSize as Add<U1>>::Output: Add<T::BlockSize> + ArrayLength<u8>,
+    <<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output: Add<U1> + ArrayLength<u8>,
+    <<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output: ArrayLength<u8> + Sub<T::SerializedStateSize>,
+    <<<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output as Sub<T::SerializedStateSize>>::Output: Sub<U1> + ArrayLength<u8>,
+    <<<<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output as Sub<T::SerializedStateSize>>::Output as Sub<U1>>::Output: Sub<T::BlockSize> + ArrayLength<u8>,
+    <<<<<<T::SerializedStateSize as Add<U1>>::Output as Add<T::BlockSize>>::Output as Add<U1>>::Output as Sub<T::SerializedStateSize>>::Output as Sub<U1>>::Output as Sub<T::BlockSize>>::Output: ArrayLength<u8>,
```

Comment:
I expected something similar, but still... :)

I think you can simplify it a fair bit by using [operator aliases](https://docs.rs/typenum/latest/typenum/operator_aliases/index.html) in bounds as well, e.g. you can replace:
```rust
<T::SerializedStateSize as Add<U1>>::Output: Add<T::BlockSize> + ArrayLength<u8>,
```
with:
```rust
Sum<T::SerializedStateSize, U1>: Add<T::BlockSize> + ArrayLength<u8>,
```